src/stream/src/from_proto/stream_scan.rs

```diff
@@ -104,16 +71,16 @@ impl ExecutorBuilder for ChainExecutorBuilder {
                     .map(|(k, v)| (k.clone(), v.clone()))
                     .collect();
                 let table_type = CdcTableType::from_properties(&properties);
-                let table_reader =
-                    table_type.create_table_reader(properties.clone(), schema.clone())?;
+                let table_reader = table_type
+                    .create_table_reader(properties.clone(), params.info.schema.clone())?;
```

Comment:
May I know where we derive the `info.schema`?

Reply:
It's from https://github.com/risingwavelabs/risingwave/blob/a288378387699a09f9e7b50b193f6e2bc8cc8eb4/src/frontend/src/optimizer/plan_node/stream_table_scan.rs#L350-L351

The `self.schema()` is derived at https://github.com/risingwavelabs/risingwave/blob/a288378387699a09f9e7b50b193f6e2bc8cc8eb4/src/frontend/src/optimizer/plan_node/generic/scan.rs#L374-L385

Reply:
I adjusted the cdc backfill executor construction logic, should be more reasonable now. PTAL again🥰